src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -159,12 +227,29 @@ var Marathon = React.createClass({
     }
   },
 
+  handleSetAppView: function (setAppView) {
```

Comment:
Why is this here along with passing the router down? Shouldn't it all come from the top?